src/components/textFields/AdvancedOpenTextField.js

```diff
@@ -1,239 +1,272 @@
+			setIsInvalid(true);
+		}
+		switch (currId) {
+			case Constants.VIN_ID: // VIN question id
+				if (!vinValidator.validate(value)) {
+					setInvalidVin(true);
+				} else {
+					setInvalidVin(false);
+				}
+				break;
+			default:
+				break;
+		}
+		// Eventually the vin switch will live inside the TextFieldValidation Area
+	};
+
+	const valueSet = (currId) => {
+		try {
+			if (existingData[currId] != null) {
+				return existingData[currId];
+			}
+		} catch (err) {
+			return null;
+		}
+	};
+
+	const onImportMapPress = () => {
+		//Sends user to map tool
+		let currStateVal = !reducerData?.response ? null : reducerData.response;
+
+		pageChange("Map", { id: id, questionID: props.data.id });
+	};
+
+	const onImportCameraPress = () => {
+		//Sends user to camera tool (for VIN, LICENSE PLATE, or DRIVERS LICENSE photo)
+
+		let type;
+		switch (importFrom) {
+			case "VINCamera":
+				type = Constants.VIN;
+				break;
+			case "plateCamera":
+				type = Constants.PLATE;
+				break;
+			case "driverIDCamera":
+				type = Constants.LICENSE;
+				break;
+			default:
+				type = Constants.VIN;
+		}
+		pageChange("Scan", {
+			objectID: id,
+			type: type,
+			questionId: currId,
+			questionReducer: questionReducer,
+		});
+	};
+
+	const onImportTimePress = () => {
+		//Gets the current date and time for user and autofills question
+
+		let now = new Date();
+		let year = now.getFullYear().toString();
+		let month = (now.getMonth() + 1).toString();
+		let date = now.getDate().toString();
+		let hours = now.getHours().toString();
+		let min = now.getMinutes().toString();
+		let fullDate =
+			year + "/" + month + "/" + date + "\t" + hours + ":" + min;
+		updateRoad({ id, question: data.humanReadableId, selection: fullDate });
+	};
+
+	const MapIcon = () => <Icon name="map-outline" color="white" size={25} />;
+
+	const CameraIcon = () => (
+		<Icon name="camera-outline" color="white" size={25} />
+	);
+
+	const ClockIcon = () => (
+		<Icon name="clock-outline" color="white" size={25} />
+	);
+
+	const RenderHeaderIcon = () => {
+		switch (importFrom) {
+			case "map":
+				return (
+					<IconButton
+						leftMargin={2}
+						icon={MapIcon()}
+						onPress={() => onImportMapPress()}
+					/>
+				);
+			case "driverIDCamera":
+				return (
+					<IconButton
+						leftMargin={2}
+						icon={CameraIcon()}
+						onPress={() => onImportCameraPress()}
+					/>
+				);
+			case "plateCamera":
+				return (
+					<IconButton
+						leftMargin={2}
+						icon={CameraIcon()}
+						onPress={() => onImportCameraPress()}
+					/>
+				);
+			case "VINCamera":
+				return (
+					<IconButton
+						leftMargin={2}
+						icon={CameraIcon()}
+						onPress={() => onImportCameraPress()}
+					/>
+				);
+			case "time":
+				return (
+					<IconButton
+						leftMargin={2}
+						icon={ClockIcon()}
+						onPress={() => onImportTimePress()}
+					/>
+				);
+		}
+	};
+
+	const ErrorMsg = () => {
+		if (invalidLength) {
+			return "Too long!";
```

Comment:
isn't it possible to also have an input that is too short? i think i need some clarification on how the invalid length is checked